git/git.go

```diff
@@ -1092,3 +1092,43 @@ func GetFilesChanged(from, to string) ([]string, error) {
 
 	return files, err
 }
+
+// IsFileModified returns whether the filepath specified is modified according
+// to `git status`. A file is modified if it has uncommitted changes in the
+// working copy or the index. This includes being untracked.
+func IsFileModified(filepath string) (bool, error) {
+
+	args := []string{
+		"-c", "core.quotepath=false", // handle special chars in filenames
+		"status",
+		"--porcelain",
+		"--", // separator in case filename ambiguous
+		filepath,
+	}
+	cmd := subprocess.ExecCommand("git", args...)
+	outp, err := cmd.StdoutPipe()
+	if err != nil {
+		return false, fmt.Errorf("Failed to call git status: %v", err)
```

Comment:
I don't think that's necessary, the `git-lfs/errors` package is a suitable replacement for stdlib `errors`.